nvtabular/io/parquet.py

```diff
@@ -109,18 +110,33 @@ def num_rows(self):
                 num_rows += piece.get_metadata().num_rows
             return num_rows
 
-    def to_ddf(self, columns=None, cpu=False):
+    def to_ddf(self, columns=None, cpu=None):
+
+        # Check if we are using cpu
+        cpu = self.cpu if cpu is None else cpu
+
         if cpu:
             # Return a Dask-Dataframe in CPU memory
-            return dd.read_parquet(
-                self.paths,
-                engine="pyarrow-dataset",
-                columns=columns,
-                index=None if columns is None else False,
-                gather_statistics=False,
-                split_row_groups=self.row_groups_per_part,
-                storage_options=self.storage_options,
-            )
+            try:
+                return dd.read_parquet(
+                    self.paths,
+                    engine="pyarrow-dataset",
+                    columns=columns,
+                    index=None if columns is None else False,
+                    gather_statistics=False,
+                    split_row_groups=self.row_groups_per_part,
+                    storage_options=self.storage_options,
+                )
+            except ValueError:
+                return dd.read_parquet(
+                    self.paths,
+                    engine="pyarrow",
+                    columns=columns,
+                    index=None if columns is None else False,
+                    gather_statistics=False,
+                    split_row_groups=self.row_groups_per_part,
+                    storage_options=self.storage_options,
+                )
```

Comment:
What's the cause of this fallback? Is this due to different versions of dask? If so - can you add a comment here explaining why this is necessary

(I'm not against having this code here - just want to make sure I understand why its there =)

Reply:
Right - Older versions of Dask will not support the `"pyarrow-dataset"` engine.  This code is indeed confusing, so I tried to clean it up a bit.